src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs

```diff
@@ -786,5 +786,85 @@ public void TestMergeChanges_NoMiddleMan()
         }
+
+        [Fact]
+        [WorkItem(26305, "https://github.com/dotnet/roslyn/issues/26305")]
+        public void WithChangesDeleteOverlap()
+        {
+            var sourceText = SourceText.From("Hello World");
+
+            var changedText = sourceText
+                .WithChanges(new TextChange(new TextSpan(5, 1), "_")) // "Hello_World"
+                .WithChanges(new TextChange(new TextSpan(6, 0), "+")); // "Hello_+World"
+
+            var changedText2 = sourceText.WithChanges(
+                new TextChange(new TextSpan(5, 1), "_"), // "Hello_World"
+                new TextChange(new TextSpan(6, 0), "+")); // "Hello_+World"
+
+            var changedText1 = sourceText
+                .WithChanges(new TextChange(new TextSpan(6, 0), "+"))
+                .WithChanges(new TextChange(new TextSpan(5, 1), "_"));
+
+            var changeList = changedText.GetTextChanges(sourceText);
+            Assert.Equal(1, changeList.Count);
+            Assert.Equal(new TextSpan(5, 1), changeList[0].Span);
+            Assert.Equal("_+", changeList[0].NewText);
+            Assert.Equal("Hello_+World", changedText.ToString());
+        }
+
+        [Fact]
+        [WorkItem(26305, "https://github.com/dotnet/roslyn/issues/26305")]
+        public void WithChanges()
+        {
+            var text0 = SourceText.From(@"using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsoleApp22
+{
+    class Class1
+    {
+        void Method()
+        {
+            Console
+
+     
+        }
+
+    }
+}");
+
+            var text1 = text0.WithChanges(new TextChange(TextSpan.FromBounds(221, 228), ""));
+            var text2 = text1.WithChanges(new TextChange(TextSpan.FromBounds(228, 229), ""));
+            var text3 = text2.WithChanges(new TextChange(TextSpan.FromBounds(228, 228), "."));
+            var text4 = text3.WithChanges(new TextChange(TextSpan.FromBounds(229, 229), "w"));
+            var text5 = text4.WithChanges(new TextChange(TextSpan.FromBounds(230, 230), "R"));
+            var text6 = text5.WithChanges(new TextChange(TextSpan.FromBounds(231, 231), "I"));
+            var text7 = text6.WithChanges(new TextChange(TextSpan.FromBounds(232, 232), "T"));
+            var text8 = text7.WithChanges(new TextChange(TextSpan.FromBounds(233, 233), "E"));
+            var text9 = text8.WithChanges(new TextChange(TextSpan.FromBounds(231, 232), ""));
+
+            var textb = text0
+                .WithChanges(new TextChange(TextSpan.FromBounds(228, 229), ""))
+                .WithChanges(new TextChange(TextSpan.FromBounds(228, 228), "."))
+                .WithChanges(new TextChange(TextSpan.FromBounds(229, 229), "w"));
+
+            var changesb = textb
+                .GetTextChanges(text0)
+                .ToList();
+
+            var textc = text0.WithChanges(
+                new TextChange(TextSpan.FromBounds(228, 229), ""),
```

Comment:
I just want to point out here that this particular sequence won't cause Merge to be called, since there is only one change list between text0 and textc.  A series of WithChanges calls and a single list of changes are two very different things. (They differ in what the positions mean.)

Reply:
Makes sense. Thanks.